enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1120 @@
+  Historical note: At one time, MicroShift relied on the OpenAPI schema for server-side validation of the Route CRD, and for this reason, kubebuilder validation tags were added to the
+  Route CRD (see <https://github.com/openshift/api/pull/1228>).
+
+  Since then, an admission plugin has been added to kube-apiserver that validates the Route CRD using the same
+  library-go validation on MicroShift that openshift-apiserver uses on OpenShift (see <https://github.com/openshift/kubernetes/pull/1416>).
+  As a result, the kubebuilder validation tags on the Route CRD are no longer required specifically for MicroShift; however, the kubebuilder validation tags are still required for client-side validation.
+
+#### Why don't you set/delete the headers on frontend public_ssl in the haproxy.config?
+  Answer: The connection is still encrypted in public_ssl, so it is not possible to examine or modify the HTTP session there. The fe_sni and fe_no_sni frontends terminate TLS.
+
+#### Can we use conditional values for headers?
+  Answer: No.  You cannot set a conditional value.
+  
+  For example: If you set x-frame-options: "%[res.hdr(server)] if { req.hdr(user-agent) -m sub evil}" on an 
+  HTTP x-frame-options response header, then the response value will be
+  the literal:  `x-frame-options: Apache/2.4.37 (centos) if { req.hdr(user-agent) -m sub evil}`. 
+  
+  In other words, the condition is not evaluated.
+  
+  All values are escaped and quoted using the single quotes. Please check section `Alternative quoting strategies for haproxy.config` for more
+  details. So the value we get after escape and quote we get is a string. 
+
+#### Why is it okay to still allow the host header on the route?
+  Answer: Custom headers that are specified on the route get configured on the corresponding backend, and the backend configuration takes effect after the frontend has done its map lookup.
+
+#### Why do we limit to max 20 items each for custom response and request headers ?
+  Answer: We selected 20 items as this is a max value we can [scale](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/2876-crd-validation-expression-language#estimated-cost-limits) 
+  w.r.t header value max size 16,384 bytes along with the CEL conditions defined.
+  
+  Setting a non-scalable value results in error [`Forbidden: estimated rule cost exceeds budget by factor of 2.7x (try simplifying the rule, or adding maxItems, maxProperties,
+  and maxLength where arrays, maps, and strings are declared)`](https://github.com/openshift/kubernetes/blob/af29f64acbf9e6d33cde7f3194cf858f54058d3b/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L1113)
+  while running integration tests.
+
+### Why did we select max value size of 16384 for header value ?
```

Comment:
This fixes the formatting so it lines up with previous and next question.
```suggestion
#### Why did we select max value size of 16384 for header value ?
```

Reply:
done